src/hooks/useOutsideClickHandler.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react'
+
+export function useOutsideClickHandler(node: HTMLDivElement | null, onOutsideClick: () => void) {
+  const handleClickOutside = (event: MouseEvent) => {
+    if (node && !node.contains(event.target as Node)) {
+      onOutsideClick()
+    }
+  }
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside)
```

Comment:
Do you have concerns that many uses of this hook will cause a lot of global mousedown handlers and eventual performance issues?

Reply:
tbh, i let copilot write this for me 😬   ..... i can optimize it for sure